### PR TITLE
implemented user login

### PR DESCRIPTION
**Pull request by meysdays**

### CODE REVIEW

**Comment by meysdays** [comment]
https://github.com/hngprojects/hng_boilerplate_nextjs/issues/314#issue-2420654584

File: src/main/java/hng_java_boilerplate/config/ApplicationConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
+    @Autowired
+    private UserRepository userRepository;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/config/ApplicationConfig.java:22]: remove autowired.

Do this instead.

```suggestion
    private final UserRepository userRepository;
```

File: src/main/java/hng_java_boilerplate/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Autowired
+    private AuthenticationProvider authenticationProvider;
+    @Autowired
+    private JwtAuthFilter jwtAuthFilter;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/config/SecurityConfig.java:22]:
```suggestion
    private  final JwtAuthFilter jwtAuthFilter;
```
Add @RequiredArgsConstructor

Make both final

File: src/main/java/hng_java_boilerplate/jwt/JwtAuthFilter.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private JwtService jwtService;
+    @Autowired
+    private UserDetailsService userDetailsService;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/jwt/JwtAuthFilter.java:27]: Use final please

File: src/main/java/hng_java_boilerplate/user/controller/AuthController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private MyUserDetailsService myUserDetailsService;
+    @Autowired
+    private JwtService jwtService;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+    @Autowired
+    private UserRepository userRepository;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/user/controller/AuthController.java:33]: Use final, import @RequiredArgsConstructor

File: src/main/java/hng_java_boilerplate/user/dto/LogInDto.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package hng_java_boilerplate.user.dto;
+
+import jakarta.validation.constraints.NotBlank;
+
+public class LogInDto {
+    @NotBlank(message = "Email and password are required.")
+    private String email;
+    @NotBlank(message = "Email and password are required.")
+    private String password;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/user/dto/LogInDto.java:9]: Separate the error message. Email is required. Password is required

Comment (billmal071) [reply src/main/java/hng_java_boilerplate/user/dto/LogInDto.java:9]: You might as well use lomboks getters and setters

File: src/main/java/hng_java_boilerplate/user/service/MyUserDetailsService.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MyUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository repository;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/user/service/MyUserDetailsService.java:15]: Same here use final, annotate the class with @RequiredArgsConstructor

File: src/main/java/hng_java_boilerplate/user/service/UserService.java [hunk 7]
```diff
@@ -14,6 +20,12 @@
 @RequiredArgsConstructor
 public class UserService {
     private final UserRepository userRepository;
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    @Autowired
+    private MyUserDetailsService myUserDetailsService;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
```
Comment (billmal071) [inline src/main/java/hng_java_boilerplate/user/service/UserService.java:28]: final

**Review by billmal071 — CHANGES_REQUESTED** [review]
Fix

File: src/main/java/hng_java_boilerplate/user/dto/LogInDto.java (hunk 5 above)
Comment (meysdays) [reply src/main/java/hng_java_boilerplate/user/dto/LogInDto.java:9]: for security reasons, being specific on what field is wrong, is that not bad?

File: src/main/java/hng_java_boilerplate/user/service/UserService.java (hunk 7 above)
Comment (meysdays) [reply src/main/java/hng_java_boilerplate/user/service/UserService.java:28]: okay

**Comment by meysdays** [comment]
fixed annotations

**Review by meysdays — COMMENTED** [review]
please review again

**Pull request closed without merging**